crates/ironrdp-pdu/src/rdp/client_info.rs

```diff
@@ -231,15 +231,111 @@ impl ExtendedClientInfo {
     }
 }
 
+pub mod extended_client_optional_info {
+    use super::*;
+    use std::marker::PhantomData;
+
+    pub struct BuilderStateSetTimeZone;
+    pub struct BuilderStateSetSessionId;
+    pub struct BuilderStateSetPerformanceFlags;
+    pub struct BuilderStateSetReconnectCookie;
+    pub struct BuilderStateFinal;
+
+    // State machine-based builder for [`ExtendedClientOptionalInfo`].
+    //
+    // [`ExtendedClientOptionalInfo`] strictly requires to set all preceding optional fields before
+    // setting the next one, therefore we use a state machine to enforce this during the compile time.
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    pub struct Builder<State> {
```

Comment:
suggestion: In case we add additional builders, in the future, it’s best to give a more specific name than just "Builder".
```suggestion
    pub struct ExtendedClientOptionalInfoBuilder<State> {
```